tokens/usecase/pricing/coingecko/pricing_coingecko_test.go

```diff
@@ -0,0 +1,117 @@
+package coingeckopricing_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/osmosis-labs/sqs/domain"
+	"github.com/osmosis-labs/sqs/domain/mocks"
+	"github.com/osmosis-labs/sqs/router/usecase/routertesting"
+	coingeckopricing "github.com/osmosis-labs/sqs/tokens/usecase/pricing/coingecko"
+	"github.com/stretchr/testify/suite"
+)
+
+type CoingeckoPricingTestSuite struct {
+	routertesting.RouterTestHelper
+}
+
+var (
+	USDC                 = routertesting.USDC
+	USDT                 = routertesting.USDT
+	ATOM                 = routertesting.ATOM
+	ETH                  = routertesting.ETH
+	defaultRouterConfig  = routertesting.DefaultRouterConfig
+	defaultPricingConfig = routertesting.DefaultPricingConfig
+)
+
+// Set up mainnet mock state with default router and pools usecase.
+func (s *CoingeckoPricingTestSuite) SetupDefaultRouterAndPoolsUsecase() routertesting.MockMainnetUsecase {
+	mainnetState := s.SetupMainnetState()
+	mainnetUsecase := s.SetupRouterAndPoolsUsecase(mainnetState, routertesting.WithRouterConfig(defaultRouterConfig), routertesting.WithPricingConfig(defaultPricingConfig))
+	return mainnetUsecase
+}
+
+func TestCoingeckoPricingTestSuite(t *testing.T) {
+	suite.Run(t, new(CoingeckoPricingTestSuite))
+}
+
+// TestGetPrices tests the GetPrice method of CoingeckoPricing.
+// MockCoingeckoPriceGetter is used to mock the CoingeckoPriceGetterFn function
+// The main goal of this test is to check if the GetPrice method returns 0 and error given non-supported base or quote denom
+func (s *CoingeckoPricingTestSuite) TestGetPrices() {
+
+	mainnetUsecase := s.SetupDefaultRouterAndPoolsUsecase()
+	defaultPricingConfig.DefaultSource = domain.CoinGeckoPricingSourceType
+	coingeckoPricingSource := coingeckopricing.New(mainnetUsecase.Router, mainnetUsecase.Tokens, defaultPricingConfig, mocks.DefaultMockCoingeckoPriceGetter)
+
+	tests := []struct {
+		desc       string
+		baseDenom  string
+		quoteDenom string
+		shouldZero bool
+		shouldErr  bool
+	}{
+		{"Test coingecko GetPrice with quote denom as USDC", ATOM, USDC, false, false},
+		{"Test coingecko GetPrice with quote denom as USDT", ATOM, USDT, false, false},
+		{"Test coingecko GetPrice with quote denom as empty string", ATOM, "", false, false},
+		{"Test coingecko GetPrice with quote denom as some spaces", ATOM, " ", false, false},
+		{"Test coingecko GetPrice with quote denom as ATOM", ATOM, ETH, true, true},
+		{"Test coingecko GetPrice with invalid base denom", "-DUMMY-", USDC, true, true},
+		{"Test coingecko GetPrice with empty base denom", "", USDC, true, true},
+		{"Test coingecko GetPrice with some spaces as base denom", " ", USDC, true, true},
+	}
+
+	for _, tt := range tests {
+		s.Run(tt.desc, func() {
+			price, err := coingeckoPricingSource.GetPrice(context.Background(), tt.baseDenom, tt.quoteDenom)
```

Comment:
Is it possible to also do some assertions on the value of `price`?

For example, on success cases, we could set up the mock in a way where if "ATOM" and "USDC" are given , the returned price is 0.5 but if it is any other denom, it would return the default of 1.